src/Sentry/Internal/Hub.cs

```diff
@@ -177,6 +178,27 @@ public void BindException(Exception exception, ISpan span)
 
         public void StartSession()
         {
+            // Attempt to recover persisted session left over from previous run
+            if (Interlocked.Exchange(ref _isPersistedSessionRecovered, 1) != 1)
+            {
+                try
+                {
+                    var recoveredSessionUpdate = _sessionManager.TryRecoverPersistedSession();
+                    if (recoveredSessionUpdate is not null)
+                    {
+                        CaptureSession(recoveredSessionUpdate);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _options.DiagnosticLogger?.LogError(
+                        "Failed to recover persisted session.",
+                        ex
+                    );
+                }
+            }
```

Comment:
I put the recovery in `StartSession()` to avoid adding a new method to `IHub`. An alternative would be to put it in the constructor though, but I think it's better to avoid I/O in there. Thoughts?